Feature - Generate Pyramid

CThompson01 opened the pull request:
Changed the pyramid of pegs to be generated at runtime to allow for dynamic sizing of pyramids in future updates. Also made balls bounce off of walls instead of being deleted when coming into contact with them.

CThompson01 committed:
Generate Pyramid

Generate the pyramid at runtime to allow for dynamic pyramid sizes in the future. Currently the pyramid offsets are not centered.

CThompson01 committed:
Make Balls Bounce Off Walls

Made balls bounce off the edges of the window now instead of just getting deleted when they go offscreen.

CThompson01 committed:
Fix Wall Bouncing

Fix wall bouncing to occur when the edge of a ball hits the edge of the screen rather than bouncing when the center of the ball hits the edge of the screen.

CThompson01 committed:
Generate in Pyramid Shape

Fixed the pyramid generation to generate in a symmetrical way now. Added 5 more pegs by default.

bqrichards commented on src/global.c:
```diff
@@ -2,4 +2,13 @@
+}
+
+unsigned int SumTo(unsigned int n) {
+    int sum = 0;
```
You may be able to change this algorithm to use the formula `n(n+1)/2` to calculate the sum of the first `n` natural numbers.

CThompson01 committed:
Change SumTo Implementation

Changed the implementation of the SumTo function to use an algorithmic approach.

CThompson01 committed:
Fixed SumTo Algorithm

Simplified the algorithm as to not have a needless extra addition calculation.
